### PR TITLE
Validate builtin interface block match

### LINKED ISSUE #702: Validate builtin interface block match
CTS has accidentally had mismatches in builtin interface between VS output and GS input. Need to validate that the block declarations are equivalent.

### PR DESCRIPTION
**Pull request by asuonpaa**
Tests and layer implementation for checking builtin blocks match between stages.

Fixes #702

### CODE REVIEW

**Comment by mark-lunarg** [comment]
LGTM (thanks @asuonpaa !) -- I would ask that before this gets pushed the order of commits be reversed -- bisecting on the first commit would break the build.

**Comment by asuonpaa** [comment]
Any action still required from me?

File: layers/shader_validation.cpp [hunk 1]
```diff
@@ -705,6 +705,83 @@ static std::map<location_t, interface_var> CollectInterfaceByLocation(shader_mod
...
+                    }
+                    case spv::DecorationBuiltIn:
+                        builtinDecorations.push_back(insn.word(1));
+                        break;
+                    default:
+                        break;
+                }
+                break;
+            // Find all input or output variables depending on the selected storage class
+            case spv::OpVariable: {
```
Comment (chrisforbes) [inline layers/shader_validation.cpp]: This must consider only variables listed in the entrypoint. There may be all kinds of other junk in the module.

File: layers/shader_validation.cpp [hunk 2]
```diff
@@ -2153,6 +2240,32 @@ static bool ValidateInterfaceBetweenStages(debug_report_data const *report_data,
         }
     }
 
+    auto builtins_producer = CollectBuiltinBlockMembers(producer, producer_entrypoint, spv::StorageClassOutput);
+    auto builtins_consumer = CollectBuiltinBlockMembers(consumer, consumer_entrypoint, spv::StorageClassInput);
```
Comment (chrisforbes) [inline layers/shader_validation.cpp]: Need to skip this if the consumer is the fragment shader

> At an interface between two non-fragment shader stages, the built-in interface block must match exactly

Comment (asuonpaa) [reply layers/shader_validation.cpp]: done

**Comment by mark-lunarg** [comment]
I'll run it through CI.

**Comment by mark-lunarg** [comment]
Tested, rebased, and pushed to Master.  Thanks!

**Pull request closed without merging**